### ISSUE TITLE
propType [name] is not required, but has no corresponding defaultProp declaration

### ISSUE THREAD

**Issue by erica93k** [yannickcr/eslint-plugin-react#1433]
I am using Flow for type checking, and I have something like this:

```
type Props = {
  name?: ?string, // this means that "name" is NOT mandatory, and its value can be **null** or **undefined**
};
```

So, my question is, should I get this linting error? Is this the desired behavior?

**Comment by jseminck** [comment]
I would say this is as expected. Since the prop is not required, you should specify a default value. You can decide to set it to `null` or `undefined` by default.

**Comment by erica93k** [comment]
I see. Thank you

**Comment by AlexanderNaiden** [comment]
I didn't understand  what should I do in my case (it's alos has eslint error propType "isSmall" is not required, but has no corresponding defaultProp declaration. (react/require-default-props)):

type SwitcherProps = {
    isChecked: boolean,
    action: () => void,
    isSmall?: boolean
}

export const Switcher = (props: SwitcherProps) => {
    return (
        <SwitcherStyle>
        <label className={"switcher ".concat(props.isSmall ? "small" : "")}>
            <input type="checkbox" checked={props.isChecked} className="input" />
            <span role="presentation" className="slider" onClick={props.action} />
        </label>
        </SwitcherStyle>
    );
};

**Comment by ljharb** [comment]
@AlexanderNaiden add:
```jsx
Switcher.defaultProps = {
  isSmall: false,
};
```

**Comment by arel** [comment]
The error message is confusing because it's singular instead of plural.

**Comment by tanishbansal20** [comment]
I fixed this issue using the default prop
  static defaultProps = {
    actions: {
      function_name: () => null,
    },
  }

**Comment by ljharb** [comment]
@tanishbansal20 note that this is wrong; the object literal isn’t the same as PropTypes.shape.
